pkg/new-kernel/patches-5.10.x/0022-siemens-ipc127-leds.patch

```diff
@@ -0,0 +1,253 @@
+ obj-$(CONFIG_LEDS_SUNFIRE)		+= leds-sunfire.o
+ obj-$(CONFIG_LEDS_SYSCON)		+= leds-syscon.o
+ obj-$(CONFIG_LEDS_TCA6507)		+= leds-tca6507.o
+diff --git a/drivers/leds/leds-siemens-ipc127.c b/drivers/leds/leds-siemens-ipc127.c
+index 57d157e..9ff45a3 100644
+--- /dev/null	2021-04-15 19:50:35.920000000 +0000
++++ a/drivers/leds/leds-siemens-ipc127.c	2021-04-15 22:54:26.370000000 +0000
+@@ -0,0 +1,203 @@
++/*
++ * drivers/leds/leds-siemends-ipc127.c
++ * Copyright (C) 2021 Roman Shaposhnik, rvs at apache dot org
++ * Based on the leds-apu.c
++ * Copyright (C) 2017 Alan Mizrahi, alan at mizrahi dot com dot ve
++ *
++ * Redistribution and use in source and binary forms, with or without
++ * modification, are permitted provided that the following conditions are met:
++ *
++ * 1. Redistributions of source code must retain the above copyright
++ *    notice, this list of conditions and the following disclaimer.
++ * 2. Redistributions in binary form must reproduce the above copyright
++ *    notice, this list of conditions and the following disclaimer in the
++ *    documentation and/or other materials provided with the distribution.
++ * 3. Neither the names of the copyright holders nor the names of its
++ *    contributors may be used to endorse or promote products derived from
++ *    this software without specific prior written permission.
++ *
++ * Alternatively, this software may be distributed under the terms of the
++ * GNU General Public License ("GPL") version 2 as published by the Free
++ * Software Foundation.
++ *
++ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
++ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
++ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
++ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE
++ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
++ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
++ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
++ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
++ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
++ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
++ * POSSIBILITY OF SUCH DAMAGE.
++ */
++
++#define pr_fmt(fmt) KBUILD_MODNAME ": " fmt
++
++#include <linux/dmi.h>
++#include <linux/err.h>
++#include <linux/init.h>
++#include <linux/io.h>
++#include <linux/kernel.h>
++#include <linux/leds.h>
++#include <linux/module.h>
++#include <linux/platform_device.h>
++
++#define IPC127_FCH_ACPI_MMIO_BASE 0xD0C50500
++#define IPC127_FCH_GPIO_BASE      IPC127_FCH_ACPI_MMIO_BASE /* we should have a longer window instead, but see below */
++#define IPC127_LEDON              0x00 /* FIXME: we can't seem to be able to turn it off */
++#define IPC127_LEDOFF             0x01 /* fully -- so we go between RED and GREEN        */
```

Comment:
So it will blink not by on/off but by red/green changes? That is a bit odd.

Reply:
Still working on it ;-) Stay tuned.